wagtail_newsletter/models.py

```diff
@@ -12,7 +17,98 @@ class Meta:  # type: ignore
     def __str__(self):
         return self.name
 
+    @property
+    def member_count(self):
+        if self.segment:
+            try:
+                return audiences.AudienceSegment.objects.get(
+                    pk=self.segment
+                ).member_count
+            except audiences.AudienceSegment.DoesNotExist:
+                return None
+
+        elif self.audience:
```

Comment:
it looks like "segment" has priority over "audience"; I reckon if the Model shouldn't force the user to only set one OR another of these 2 fields - in order to avoid unexpected behaviours when both would be set but only the "segment" is taken into account? 
(in its `clean` method for example)

Reply:
Well, `audience` is a required field, and the values for `segment` are chosen from the selected audience's segments. This got me thinking to test what happens to the segment picker if no audience was selected (it crashed; fixed it). Also, we should probably validate that the `segment` is indeed part of the `audience` on save, lest the user change the audience and leave a stale segment selected.

Reply:
Thanks for the explanation! :slightly_smiling_face:

Reply:
I've added validation to make sure that `segment` is among the segments of the chosen `audience`.